### PR TITLE
[TX flow] feat: replace owner

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves
Implements "Replace Owner" designs for the new tx flow

## How to test it
Navigate the "Replace Owner" flow

## Screenshots
Screenshot 2023-06-22 at 11 08 36
Screenshot 2023-06-22 at 14 44 31

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/common/NameInput/index.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ const NameInput = ({
       error={Boolean(fieldError)}
       fullWidth
       required={required}
-      sx={{ '& .MuiInputBase-input': { p: 3, px: 2 } }}
+      sx={{ mt: 1, '& .MuiInputBase-input': { py: '22px', px: 2 } }}
```
Comment (iamacook) [inline src/components/common/NameInput/index.tsx:29]: Nit: let's move this to the CSS module.

File: src/components/tx-flow/flows/AddOwner/ChooseOwner.tsx [hunk 2]
```diff
@@ -66,20 +66,22 @@ export const ChooseOwner = ({
...
 
-  const newNumberOfOwners = safe.owners.length + 1
+  const newNumberOfOwners = safe.owners.length + (!params.removedOwner ? 1 : 0)
 
   return (
     <TxCard>
       <FormProvider {...formMethods}>
         <form onSubmit={onFormSubmit} className={commonCss.form}>
-          <Box mb={1}>
+          <Typography variant="body2" mb={1}>
```
Comment (iamacook) [inline src/components/tx-flow/flows/AddOwner/ChooseOwner.tsx:75]: The `Typography` element should also be conditional.

File: src/components/tx-flow/flows/ReplaceOwner/index.tsx [hunk 3]
```diff
@@ -32,7 +33,7 @@ const ReplaceOwnerFlow = ({ address }: { address: string }) => {
   ]
 
   return (
-    <TxLayout title="Replace owner" step={step} onBack={prevStep}>
+    <TxLayout title="New transaction" subtitle="Replace owner" icon={SaveAddressIcon} step={step} onBack={prevStep}>
```
Comment (iamacook) [inline src/components/tx-flow/flows/ReplaceOwner/index.tsx:36]:
```suggestion
    <TxLayout title={step === 0 ? "New transaction" : "Confirm transaction"} subtitle="Replace owner" icon={SaveAddressIcon} step={step} onBack={prevStep}>
```

**Review by iamacook — CHANGES_REQUESTED** [review]
You can now delete the steps in the review owner flow folder, the `SetThreshold` step of the add owner flow and remove any unnecessary CSS as well.

**Comment by katspaugh** [comment]
Belated design feedback, this is really hard to parse:
Screenshot 2023-06-22 at 11 26 52

I would expect it to look more like:
* Removed owner: 0x1243
* Added owner: 0x5678

File: src/components/tx-flow/flows/AddOwner/ReviewOwner.tsx [hunk 4]
```diff
@@ -57,8 +58,26 @@ export const ReviewOwner = ({ params }: { params: AddOwnerFlowProps | ReplaceOwn
       <Paper sx={{ backgroundColor: ({ palette }) => palette.background.main, p: '12px' }}>
         <EthHashInfo name={addressBook[safeAddress]} address={safeAddress} shortAddress={false} />
```
Comment (iamacook) [inline src/components/tx-flow/flows/AddOwner/ReviewOwner.tsx]: The label ("Selected Safe Account") [is missing](https://www.figma.com/file/ptTs6lDBeUuLNySroJ5PiF/Web-Master-File?type=design&node-id=7057-169722&mode=design&t=aZRJCFMNH1sCOvcR-0). I'd also check how this

File: src/components/tx-flow/flows/AddOwner/ReviewOwner.tsx [hunk 5]
```diff
@@ -54,16 +55,55 @@ export const ReviewOwner = ({ params }: { params: AddOwnerFlowProps | ReplaceOwn
...
+        </Typography>
+        <Paper sx={{ backgroundColor: ({ palette }) => palette.background.main, p: '12px' }}>
+          <EthHashInfo name={addressBook[safeAddress]} address={safeAddress} shortAddress={false} />
+        </Paper>
+      </div>
+      {params.removedOwner && (
+        <>
+          <Box display="flex" alignItems="center" gap={2} mx="auto">
+            <SvgIcon component={MinusIcon} inheritViewBox fontSize="small" />
+            Remove the owner
```
Comment (iamacook) [inline src/components/tx-flow/flows/AddOwner/ReviewOwner.tsx:70]:
```suggestion
            Remove owner
```

**Review by iamacook — APPROVED** [review]
